superset/db_engine_specs/postgres.py

```diff
@@ -45,6 +47,96 @@ class PostgresBaseEngineSpec(BaseEngineSpec):
     engine = ""
     engine_name = "PostgreSQL"
 
+    column_type_mappings = (
+        (
+            re.compile(r"^smallint", re.IGNORECASE),
+            types.SMALLINT,
+            utils.GenericDataType.NUMERIC,
```

Comment:
Nit: can we import `GenericDataType` directly and get rid of the `utils.` prefix? Just want the code the look cleaner.

Reply:
But, is adding `GenericDataType` here really necessary, though? I'd imagine all SQLA types could be definitively mapped to a `GenericDataType`, wouldn't they? Maybe `ColumnSpec` can just have an instance `@property` to map all known SQLA types to `GenericDataType`?

Reply:
@ktmud Yes, it could've been mapped as you proposed. The reason behind the current way is that if we get multiple matches we can prioritise those at the start of the list without necessarily sorting the results for the most part.

Reply:
I understand the need to use a list to have priority for RegExp matching, but will there be a case where two SQLA types may be matched to different `GenericDataType`? If not, isn't `GenericDataType` already inferred by SQLA `types`?